packages/airnode-validator/src/ois/ois.ts

```diff
@@ -106,11 +113,50 @@ export const operationSchema = z.object({
   parameters: z.array(operationParameterSchema),
 });
 
-export const pathSchema = z.record(operationSchema);
+export const httpStatusCodes = z.union([z.literal('get'), z.literal('post')]);
+
+export const pathSchema = z.record(httpStatusCodes, operationSchema);
+
+const ensurePathParametersExist: ValidatorRefinement<SchemaType<typeof pathsSchema>> = (paths, ctx) => {
+  forEach(paths, (pathData, rawPath) => {
+    forEach(pathData, (paramData, httpMethod) => {
+      const parameters = paramData!.parameters;
+      // Match on anything in the path that is braces
+      // i.e. The path /users/{id}/{action} will match ['{id}', '{action}']
+      const regex = /\{([^}]+)\}/g;
+      const matches = rawPath.match(regex)?.map(removeBraces) ?? [];
+
+      // Check that all path parameters are defined
+      matches.forEach((match) => {
+        const parameter = parameters.find((p) => p.in === 'path' && p.name === match);
+        if (!parameter) {
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: `Path parameter "${match}" is not found in "parameters"`,
```

Comment:
Nit: Do we want past or present tense for these kinds of errors? Maybe past since the validation has already completed when the user sees it?

```suggestion
            message: `Path parameter "${match}" was not found in "parameters"`,
```

Reply:
In my experience present is more used and it also sounds better. The way I think about it is that the validation happened in the past, but your config is invalid now (in present).

Reply:
Sure as long as we try to keep it consistent 👌🏻